frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java

```diff
@@ -1239,4 +1267,131 @@ private void updateVulnerabilityVerification(ResourceRequest request, ResourceRe
+
+        JSONObject jsonResult = createJSONObject();
+        jsonResult.put(DATATABLE_RECORDS_TOTAL, componentList.size());
+        jsonResult.put(DATATABLE_RECORDS_FILTERED, componentList.size());
+        jsonResult.put(DATATABLE_DISPLAY_DATA, jsonComponents);
+
+        try {
+            writeJSON(request, response, jsonResult);
+        } catch (IOException e) {
+            log.error("Problem rendering RequestStatus", e);
+        }
+    }
+
+    public JSONArray getComponentData(List<Component> componentList, Map<String, Integer> componentParameters) {
+        int startIdx = componentParameters.get(DATATABLE_DISPLAY_START);
+        int listDisplayAmount = componentParameters.get(DATATABLE_DISPLAY_LENGTH);
+        int sortColumnIdx = componentParameters.get(DATATABLE_SORT_COLUMN);
+        boolean isAscending = isSortOrderAscending(componentParameters);
+
+        List<Component> sortedComponents = sortComponentList(componentList, sortColumnIdx, isAscending);
+        int count = listDisplayAmount == -1 ? sortedComponents.size() : startIdx + listDisplayAmount;
+
+        JSONArray componentData = createJSONArray();
+        for (int i = startIdx; i < count; i++) {
+            if (i < sortedComponents.size()) {
+                JSONObject jsonObject = JSONFactoryUtil.createJSONObject();
+                Component comp = sortedComponents.get(i);
+                jsonObject.put("id", comp.getId());
+                jsonObject.put("DT_RowId", comp.getId());
+                jsonObject.put("name", SW360Utils.printName(comp));
+                jsonObject.put("cType", nullToEmptyString(comp.getComponentType()));
+                jsonObject.put("lRelsSize", String.valueOf(comp.getReleaseIdsSize()));
+                jsonObject.put("attsSize", String.valueOf(comp.getAttachmentsSize()));
+
+                JSONArray vendorArray = createJSONArray();
+                if (comp.isSetVendorNames()) {
+                    for (String vendor : comp.getVendorNames().stream().sorted().collect(Collectors.toList())) {
+                        vendorArray.put(vendor);
+                    }
+                } else {
+                    vendorArray.put("");
+                }
+
+                JSONArray licenseArray = createJSONArray();
+                if (comp.isSetMainLicenseIds()) {
+                    for (String license : comp.getMainLicenseIds().stream().sorted().collect(Collectors.toList())) {
+                        licenseArray.put(license);
+                    }
+                } else {
+                    licenseArray.put("");
+                }
+
+                jsonObject.put("vndrs", vendorArray);
+                jsonObject.put("lics", licenseArray);
+                componentData.put(jsonObject);
+            }
+        }
+
+        return componentData;
+    }
+
+    private boolean isSortOrderAscending(Map<String, Integer> componentParameters) {
+        if (componentParameters.get(DATATABLE_SORT_DIRECTION).equals(0)) {
```

Comment:
why not `return componentParameters.get(DATATABLE_SORT_DIRECTION).equals(0);`

Reply:
removed if statement and return `componentParameters.get(DATATABLE_SORT_DIRECTION).equals(0);` directly